shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/transaction/TransactionStatus.java

```diff
@@ -34,6 +34,9 @@
     
     @Setter
     private volatile boolean inTransaction;
+
+    @Setter
+    private volatile boolean inXA;
```

Comment:
Could we use the existing`transactionType` and `inTransaction`?

Reply:
No, I don't think so. They are two different transaction types and I think ```inXA``` is used in cases which handle the XA related statments to mark that the current transaction is in XA mode.

Reply:
If not `inTransaction`, connections will be returned to DataSource after each `CommandExecuteTask` finished. And which thread pool is selected to execute the task is related to `transactionType`(Refer to the class `OKProxyState`). I think this issue is not so simple that just passing the SQL to actual databases.

Reply:
Well, can you point out the codes related ? I can take a close look.

Reply:
You may check the following classes
- `org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask`
- `org.apache.shardingsphere.proxy.frontend.state.impl.OKProxyState`

Reply:
Thanks - I will take look.

Reply:
So I think it should change the following
https://github.com/apache/shardingsphere/blob/8fc35c9afee87627dd43381c086dfe81df062b47/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/transaction/TransactionStatus.java#L56-L64

with ```return (InTransaction && TransactionType.BASE != transactionType) || inXA;```

So we can hold the connections during the XA transaction. Does it make sense ?

Reply:
I see your point. The `TransactionType.XA` and `inXa` seem confusing. I think the `inXa` is something like `TransactionType.MANUAL`.
From perspective of performance, broadcasting the SQL may cause cause high latency, especially if there is a large quantity of actual data nodes.

Reply:
yeah, exactly - It should be user's resposiblity to control the transactions. And I can understand the performance issues,  if the bussiness logic has a strong data consistent, that must be a tradeofff.

So, can we introduce the ```TransactionType.MANUAL```? and if the user select this type, the proxy will broadcast all of the transaction related statements to the backend.

Does it make sense ?

Reply:
By the way, can you show the transaction document on shardingsphere website ? I think we must have some clear ```NOTE``` to mention customers that the different behaviors of the transaction types.

Reply:
> So I think it should change the following
> 
> https://github.com/apache/shardingsphere/blob/8fc35c9afee87627dd43381c086dfe81df062b47/shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/transaction/TransactionStatus.java#L56-L64
> 
> with `return (InTransaction && TransactionType.BASE != transactionType) || inXA;`
> 
> So we can hold the connections during the XA transaction. Does it make sense ?

@zhfeng @TeslaCN Maybe we can use
```
return inTransaction && (TransactionType.XA == transactionType || .....);
```

Reply:
Well,  I think ```TransactionType.XA``` here is different semantics with ```inXA``.

* ```TransactionType.XA``` is used by shardingsphere to manage the distributed transaction **INTERNAL**
* ```inXA``` means that users will be responsible to manage the transaction by themself, espcially using XA. So I'd like to change to use ```TransactionType.MANUAL```.

Also I'd like to understand the desgin strategy of transactions in shardingsphere in details. @terrymanu is there any document can be shared ?

Reply:
> Well, I think `TransactionType.XA` here is different semantics with ```inXA``.
> 
> * `TransactionType.XA` is used by shardingsphere to manage the distributed transaction **INTERNAL**
> * `inXA` means that users will be responsible to manage the transaction by themself, espcially using XA. So I'd like to change to use `TransactionType.MANUAL`.
> 
> Also I'd like to understand the desgin strategy of transactions in shardingsphere in details. @terrymanu is there any document can be shared ?
how about `TransactionType.XA` and `inTransaction` equals to `inXA`? @zhfeng

Reply:
No, I'd like to say them are totaly different here.